src/main/java/net/floodlightcontroller/forwarding/Forwarding.java

```diff
@@ -607,6 +608,9 @@ protected Match createMatchFromPacket(IOFSwitch sw, OFPort inPort, OFPacketIn pi
                     if (FLOWMOD_DEFAULT_MATCH_TRANSPORT_DST) {
                         mb.setExact(MatchField.TCP_DST, tcp.getDestinationPort());
                     }
+                    if(FLOWMOD_DEFAULT_MATCH_TCP_FLAG){
```

Comment:
@allewwaly you can use IOFSwitch's getDescription() to get switch hardware and software information. I'd suggest connecting to an OVS and logging what it sends (Wireshark or OVS commands would also work). I'd then pick the minimal checks to see if a switch is OVS (and ideally the version of OVS that first supported the flags).